src/tile/MediaView.test.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import { describe, expect, test } from "vitest";
+import { render, screen } from "@testing-library/react";
+import { axe } from "vitest-axe";
+import { TooltipProvider } from "@vector-im/compound-web";
+import {
+  TrackReference,
+  TrackReferencePlaceholder,
+} from "@livekit/components-core";
+import { Track, TrackPublication } from "livekit-client";
+import { type ComponentProps } from "react";
+
+import { MediaView } from "./MediaView";
+import { EncryptionStatus } from "../state/MediaViewModel";
+import { mockLocalParticipant } from "../utils/test";
+
+describe("MediaView", () => {
+  const participant = mockLocalParticipant({});
+  const trackReferencePlaceholder: TrackReferencePlaceholder = {
+    participant,
+    source: Track.Source.Camera,
+  };
+  const trackReference: TrackReference = {
+    ...trackReferencePlaceholder,
+    publication: new TrackPublication(Track.Kind.Video, "id", "name"),
+  };
+
+  const baseProps: ComponentProps<typeof MediaView> = {
+    displayName: "some name",
+    videoEnabled: true,
+    videoFit: "contain",
+    targetWidth: 300,
+    targetHeight: 200,
+    encryptionStatus: EncryptionStatus.Connecting,
+    mirror: false,
+    unencryptedWarning: false,
+    video: trackReference,
+    member: undefined,
+  };
+
+  test("is accessible", async () => {
+    const { container } = render(<MediaView {...baseProps} />);
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  describe("placeholder track", () => {
+    test("neither video nor avatar are shown", () => {
+      render(<MediaView {...baseProps} video={trackReferencePlaceholder} />);
+      expect(screen.queryByTestId("video")).toBeNull();
+      expect(screen.getByTestId("avatar")).not.toBeVisible();
```

Comment:
Can we use role-based queries for these rather than test IDs?

Reply:
I don't see a role for video elements so I am leaving that as a testid.